core/src/main/java/hudson/console/ConsoleLogFilter.java

```diff
@@ -69,7 +69,7 @@ public OutputStream decorateLogger(AbstractBuild build, OutputStream logger) thr
             // happens only if the subtype fails to override neither decorateLogger method
             throw new AssertionError("The plugin '" + this.getClass().getName() + "' still uses " +
                     "deprecated decorateLogger(AbstractBuild,OutputStream) method. " +
-                    "Update the plugin to use setUp(Run,OutputStream) instead.");
+                    "Update the plugin to use decorateLogger(Run,OutputStream) instead.");
```

Comment:
```suggestion
                    "Update the plugin to call #decorateLogger(Run,OutputStream) instead.");
```

It was unclear to me what is meant here because I first thought it was referring to what was implemented (and that seemed to contradict the comment above). "Call" would be clearer.

OTOH, why would you report the current class's name if you're interested in the caller of a public method? Something's weird here.

Reply:
Not sure I'd add the `#` because this isn't JavaDoc, but not against it either.
Adding the class name would be OK, but it would be reported next to the message in the exception's stack trace too.

As for the rest of the wording, there are many similar pieces of code similar to this (hence the copy/paste error of the method name in the first place), so if that's getting changed, it should probably be changed across the board.

Reply:
Actually, that branch is reached when a `ConsoleFilter` overrides neither method, which is logically an `AbstractMethodError()`.
So if anything, it should be "implemenet" or "override" instead of "use" or "call".
Or just drop the `AssertionError` and throw `AbstractMethodException` instead, which is what `SimpleBuildStep` does.

Whatever change is made should then also be made in `BuildWrapper` (looks like it's where this was copied from).

Reply:
> Actually, that branch is reached when a `ConsoleFilter` overrides neither method, which is logically an `AbstractMethodError()`.

Right, but then the message is just plain wrong. The mentioned method isn't overridden either.

Reply:
I've changed both places to use `AbstractMethodError` with a better message.